registration-processor/init/registration-processor-registration-status-service/src/main/java/io/mosip/registration/processor/status/api/controller/handler/RegistrationSyncExceptionHandler.java

```diff
@@ -142,13 +145,20 @@ private ResponseEntity<Object> buildRegStatusExceptionResponse(Exception ex) {
 		response.setResponsetime(DateUtils.getUTCCurrentDateTimeString(env.getProperty(DATETIME_PATTERN)));
 		response.setVersion(env.getProperty(REG_SYNC_APPLICATION_VERSION));
 		response.setResponse(null);
-		Gson gson = new GsonBuilder().serializeNulls().create();
+	//	Gson gson = new GsonBuilder().serializeNulls().create();
 
 
 		if(isEnabled) {
 			HttpHeaders headers = new HttpHeaders();
-			headers.add(RESPONSE_SIGNATURE,digitalSignatureUtility.getDigitalSignature(gson.toJson(response)));
-			return ResponseEntity.ok().headers(headers).body(gson.toJson(response));
+			String res=null;
+			try {
+				res=objectMapper.writeValueAsString(response);
+			} catch (IOException e1) {
+				regProcLogger.error("Error while processing response",e1);
```

Comment:
here since you have added catch block the exception response should be handled properly. In current even if exception happens you are sending response object which is wrong